javascript/ql/src/semmle/javascript/dataflow/TaintTracking.qll

```diff
@@ -87,22 +87,26 @@ module TaintTracking {
      */
     predicate isSanitizerGuard(SanitizerGuardNode guard) { none() }
 
-    final override predicate isBarrier(DataFlow::Node node) {
-      super.isBarrier(node) or
-      isSanitizer(node) or
-      node instanceof DataFlow::VarAccessBarrier
+    override predicate isLabeledBarrier(DataFlow::Node node, DataFlow::FlowLabel lbl) {
+      super.isLabeledBarrier(node, lbl)
+      or
+      isSanitizer(node) and lbl.isTaint()
     }
 
-    final override predicate isBarrierEdge(DataFlow::Node source, DataFlow::Node sink) {
-      super.isBarrierEdge(source, sink) or
-      isSanitizerEdge(source, sink)
+    override predicate isBarrier(DataFlow::Node node) {
```

Comment:
You removed the `final override predicate isBarrierEdge(DataFlow::Node source, DataFlow::Node sink)`. 

I think we should keep the override for consistency with the labeled variant of `isBarrierEdge` (just without calling `isSanitizerEdge`). 
E.g. 
```
final override predicate isBarrierEdge(DataFlow::Node source, DataFlow::Node sink) {	
  super.isBarrierEdge(source, sink)
}
```

Reply:
Hm, so the idea was that you could override the non-labeled version of `isBarrier` and `isBarrierEdge` to block all labels. To do the same with the labeled versions requires making a cartesian product of all flow labels, which can be expensive when there are also many barrier nodes/edges.

For example, suppose you block all nodes that occur in client-side code, and there are 8 flow labels, the factor 64 is enough to matter.